web-src/src/components/GenerateButton.js

```diff
@@ -74,12 +74,22 @@ export function GenerateButton() {
     sampleRUM('genai:prompt:generate', { source: 'GenerateButton#handleGenerate' });
     setGenerationInProgress(true);
     setIsOpenPromptEditor(false);
+    const startTime = Date.now();
     generateResults()
       .catch((error) => {
         ToastQueue.negative(error.message, { timeout: 2000 });
+
+        // capture network error (status 500)
+        if (error.status === 500) {
```

Comment:
We are capturing several error codes from the backend. Can you refer to this file (https://github.com/adobe/aem-genai-assistant/blob/68f48403b7ca801ddb89444990038ecb63740efb/actions/FirefallClient.js#L26) and rewrite this logic?